packages/data-sdk/src/connector/data/BaseUniverseDataConnector.ts

```diff
@@ -167,32 +160,38 @@ export class BasicUniverseDataConnector {
   }
 
   protected getAvailablePCDWorker(): Worker | undefined {
+    const pcdWorkerPool = WorkerPoolService.getPcdWorkerPool();
+
     for (let i = 0; i < PCD_WORKER_POOL_SIZE; i++) {
       if (!this.pcdWorkerPoolOccupancy[i]) {
         this.pcdWorkerPoolOccupancy[i] = true;
-        return this.pcdWorkerPool[i];
+        return pcdWorkerPool[i];
       }
     }
     return undefined;
   }
 
   protected getAvailableDataFetchWorker(): Worker | undefined {
+    const dataFetchWorkerPool = WorkerPoolService.getDataFetchWorkerPool();
+
     for (let i = 0; i < DATA_FETCH_WORKER_POOL_SIZE; i++) {
       if (!this.dataFetchWorkerPoolOccupancy[i]) {
         this.dataFetchWorkerPoolOccupancy[i] = true;
-        return this.dataFetchWorkerPool[i];
+        return dataFetchWorkerPool[i];
       }
     }
     return undefined;
   }
 
   protected releasePCDWorker(worker: Worker) {
-    const index = this.pcdWorkerPool.indexOf(worker);
+    const pcdWorkerPool = WorkerPoolService.getPcdWorkerPool();
+    const index = pcdWorkerPool.indexOf(worker);
     this.pcdWorkerPoolOccupancy[index] = false;
```

Comment:
I think we can call .terminate() here vs. waiting for chrome to clean it up

Reply:
@useafterfree So, poolOccupancy tracks the workers that are currently being utilized.  

In this function, we are just releasing the worker from the tracking pool.  We don't want to terminate the worker because we do want it to persist for as long as the app is running for the user.

@andrey-mazepas does this sound correct?

Reply:
I think we should using these workers in v2 views eventually, so It'd be good to have them available to manage fetching assets.
But if it's only for the scene module we could terminate and recreate them